lib/viewmodels/home_view_model.dart

```diff
@@ -1,7 +1,70 @@
 import "package:hooks_riverpod/hooks_riverpod.dart";
-import "package:proxima/models/ui/post_data.dart";
+import "package:proxima/models/ui/post_overview.dart";
+import "package:proxima/services/database/post_repository_service.dart";
+import "package:proxima/services/database/user_repository_service.dart";
+import "package:proxima/services/geolocation_service.dart";
 
-// This provider is used to store the list of posts that are displayed in the home feed.
-final postList = Provider<List<Post>>((ref) {
-  return List.empty();
-});
+/// This viewmodel is used to fetch the list of posts that are displayed in the home feed.
+/// It fetches the posts from the database and returns a list of [PostOverview] objects to be displayed.
+class HomeViewModel extends AsyncNotifier<List<PostOverview>> {
+  HomeViewModel();
+
+  static const kmPostRadius = 0.1;
+
+  @override
+  Future<List<PostOverview>> build() async {
+    final geoLocationService = ref.watch(geoLocationServiceProvider);
+    final postRepository = ref.watch(postRepositoryProvider);
+    final userRepository = ref.watch(userRepositoryProvider);
+
+    final position = await geoLocationService.getCurrentPosition();
+
+    final postsFirestore =
+        await postRepository.getNearPosts(position, kmPostRadius);
+
+    final postOwnersId =
+        postsFirestore.map((post) => post.data.ownerId).toSet();
+
+    final postOwners = await Future.wait(
+      postOwnersId.map((userId) => userRepository.getUser(userId)),
+    );
+
+    final posts = postsFirestore.map((post) {
+      final owner = postOwners.firstWhere(
+        (user) => user.uid == post.data.ownerId,
+        // This should never be executed in practice as if the owner is not found,
+        // the user repository would have already thrown an exception.
+        orElse: () => throw Exception("Owner not found"),
+      );
+
+      return PostOverview(
+        title: post.data.title,
+        description: post.data.description,
+        votes: post.data.voteScore,
+        posterUsername: owner.data.username,
+        commentNumber:
+            0, // TODO: Update appropriately when comments are implemented
+      );
+    }).toList();
+
+    return posts;
+  }
+
+  /// Refresh the list of posts
+  /// This will put the state of the viewmodel to loading, fetch the posts
+  /// and update the state accordingly
+  Future<void> refresh() async {
+    state = const AsyncValue.loading();
+    try {
+      final posts = await build();
+      state = AsyncValue.data(posts);
+    } catch (e, stackTrace) {
+      state = AsyncValue.error(e, stackTrace);
+    }
```

Comment:
https://codewithandrea.com/tips/async-value-guard-try-catch/
I'm just reading this code to get inspiration for my PR. Shouldn't a AsynValue.guard be used here?

Reply:
Yes, you're totally right.
I wasn't aware of the existence of this method so thank you for the suggestion !
Applied in commit 7bd15a62afb95316691e695dc6bb37feef40807a.